src/main/java/com/feywild/feywild/block/entity/MagicalBrazier.java

```diff
@@ -0,0 +1,57 @@
+package com.feywild.feywild.block.entity;
+
+import io.github.noeppi_noeppi.libx.base.tile.BlockEntityBase;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import software.bernie.geckolib3.core.IAnimatable;
+import software.bernie.geckolib3.core.PlayState;
+import software.bernie.geckolib3.core.builder.AnimationBuilder;
+import software.bernie.geckolib3.core.controller.AnimationController;
+import software.bernie.geckolib3.core.event.predicate.AnimationEvent;
+import software.bernie.geckolib3.core.manager.AnimationData;
+import software.bernie.geckolib3.core.manager.AnimationFactory;
+
+public class MagicalBrazier extends BlockEntityBase implements IAnimatable {
+
+    private final AnimationFactory animationFactory = new AnimationFactory(this);
+
+    private int textureNumber = 1;
+    private int animationCount = 0;
+
+    public MagicalBrazier(BlockEntityType<?> type, BlockPos pos, BlockState state) {
+        super(type, pos, state); //1.16 only has type
+    }
+
+    public int getTextureNumber() {
```

Comment:
This is a getter, it should not modify state and by doing it this way, the animation speed will change depending on the fps count.

This should use a `tick` method instead. And inside that tick method, check to only tick on the client side.